src/armnn.js

```diff
@@ -0,0 +1,636 @@
+
+        // generate parameters
+        let paramIdx = 0;
+        for (let j = 0; j < graph.layersLength(); j++) {
+            let base = armnn.Node.getBase(graph.layers(j));
+            for (let i = 0 ; i < base.outputSlotsLength() ; i++) {
+                let key = armnn.Parameter.makeKey(base.index(), i);
+                let name = paramIdx.toString();
+
+                params[key] = new armnn.Parameter(name, name, base.outputSlots(i).tensorInfo(), null);
+                paramIdx++;
+            }
+        }
+
+        // generate nodes
+        for (let j = 0; j < graph.layersLength(); j++) {
+            this._nodes.push(new armnn.Node(graph.layers(j), params, metadata));
+        }
+
+        // link inputs
+        for (let k = 0; k < graph.inputIdsLength(); k++) {
+            // need to do something?
+        }
+
+        // link outputs
+        for (let l = 0; l < graph.outputIdsLength(); l++) {
+            // need to do something?
+        }
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get groups() {
+        return false;
+    }
+
+    get inputs() {
+        return this._inputs;
+    }
+
+    get outputs() {
+        return this._outputs;
+    }
+
+    get nodes() {
+        return this._nodes;
+    }
+};
+
+armnn.Node = class {
+
+    constructor(layer, params, metadata) {
+        this._metadata = metadata;
+        this._operator = armnn.schema.LayerName[layer.layerType()];
+
+        let base = armnn.Node.getBase(layer);
+
+        this._name = '';
+        this._outputs = [];
+        this._inputs = [];
+        this._category = '';
+        this._attributes = [];
+
+        if (base) {
+            this._name = base.layerName();
+
+            for (let i = 0; i < base.inputSlotsLength(); i++) {
+                let srcConnection = base.inputSlots(i).connection();
+                let srcLayerIdx = srcConnection.sourceLayerIndex()
+                let srcOutputIdx = srcConnection.outputSlotIndex()
+                this._inputs.push(params[armnn.Parameter.makeKey(srcLayerIdx, srcOutputIdx)]);
+            }
+
+            for (let j = 0; j < base.outputSlotsLength(); j++) {
+                this._outputs.push(params[armnn.Parameter.makeKey(base.index(), j)]);
+            }
+        }
+        this.setAttribute(layer);
+    }
+
+    get operator() {
+        return this._operator;
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get domain() {
+        return null;
+    }
+
+    get documentation() {
+        return '';
+    }
+
+    get group() {
+        return null;
+    }
+
+    get category() {
+        return this._category;
+    }
+
+    get inputs() {
+        return this._inputs;
+    }
+
+    get outputs() {
+        return this._outputs;
+    }
+
+    get attributes() {
+        return this._attributes;
+    }
+
+    static castLayer(layer) {
+        let layerType = layer.layerType();
+
+        if (layerType == armnn.schema.Layer.InputLayer)
```

Comment:
`{` and `}`

Reply:
I made it manually, but since I am not expert on javascript I have no idea how to avoid manual update